pkg/qrcodegenerator/grpc_end_devices.go

```diff
@@ -19,34 +19,65 @@ import (
 
 	pbtypes "github.com/gogo/protobuf/types"
 	qrcodegen "github.com/skip2/go-qrcode"
-	"go.thethings.network/lorawan-stack/v3/pkg/qrcode"
+	clusterauth "go.thethings.network/lorawan-stack/v3/pkg/auth/cluster"
+	"go.thethings.network/lorawan-stack/v3/pkg/errors"
+	"go.thethings.network/lorawan-stack/v3/pkg/rpcmetadata"
 	"go.thethings.network/lorawan-stack/v3/pkg/ttnpb"
 )
 
+var errUnauthenticated = errors.DefineUnauthenticated("unauthenticated", "call was not authenticated")
+
 type endDeviceQRCodeGeneratorServer struct {
 	QRG *QRCodeGenerator
 }
 
+// GetFormat implements EndDeviceQRCodeGenerator.
 func (s *endDeviceQRCodeGeneratorServer) GetFormat(ctx context.Context, req *ttnpb.GetQRCodeFormatRequest) (*ttnpb.QRCodeFormat, error) {
-	format := qrcode.GetEndDeviceFormat(req.FormatId)
+	_, err := rpcmetadata.WithForwardedAuth(ctx, s.QRG.AllowInsecureForCredentials())
+	if errors.IsUnauthenticated(err) {
+		if clusterauth.Authorized(ctx) != nil {
+			return nil, errUnauthenticated.New()
+		}
```

Comment:
Don't we have a simple check to see if the caller is authenticated?

Reply:
Can you come up with something very simple in the rights package? We really only have to check the incoming context for anything non-empty in `AuthType` and `AuthValue`.

Reply:
> We really only have to check the incoming context for anything non-empty in AuthType and AuthValue.

`WithForwardedAuth` already does that actually. https://github.com/TheThingsNetwork/lorawan-stack/blob/9d58a370c158a60072ed841370a5e7a97b67f419/pkg/rpcmetadata/request.go#L40-L44

This doesn't cover ClusterAuth which is not set in the Metadata.

Should I maybe collect these lines in a single function to check "any" auth?